solr/core/src/java/org/apache/solr/cluster/events/impl/DelegatingClusterEventProducer.java

```diff
@@ -90,7 +95,10 @@ public void setDelegate(ClusterEventProducer newDelegate) {
         log.debug("--- delegate {} already in state {}", delegate, delegate.getState());
       }
     }
-    this.version++;
+    synchronized (version) {
```

Comment:
I don't think we need AtomicInteger if all sections that access `version` are synchronized?